src/main/java/olya/app/remindme/controller/UserController.java

```diff
@@ -22,25 +28,74 @@
 public class UserController {
     private final UserService userService;
     private final UserMapper userMapper;
+    private final TokenService tokenService;
 
+    //public
     @PostMapping("/register")
     public UserResponseDto register(@Valid @RequestBody UserRequestDto userRequestDto) {
         User user = userService.create(userRequestDto);
         return userMapper.mapToDto(user);
     }
 
+    //for user and admin
+    @GetMapping("/my-user")
+    public UserResponseDto getCurrentUser(@AuthenticationPrincipal UserPrincipal principal) {
+        User user = userService.getByEmail(principal.getEmail());
+        return userMapper.mapToDto(user);
+    }
+
+    //for user and admin
+    @PutMapping("/my-user")
+    public UserResponseDto updateCurrentUser(@AuthenticationPrincipal UserPrincipal principal, @Valid @RequestBody UserRequestDto userRequestDto) {
+        User user = userService.getByEmail(principal.getEmail());
+        User updatedUser = userService.update(user, userRequestDto);
+        return userMapper.mapToDto(updatedUser);
+    }
+
+    //for user and admin
+    @PutMapping("/my-user/deactivate")
+    public void deactivateCurrentUser(@AuthenticationPrincipal UserPrincipal principal) {
+        User user = userService.getByEmail(principal.getEmail());
+        userService.deactivate(user);
+    }
+
+    //for admin only
+    //get user by email
+    @GetMapping("/by-email")
+    public User getByEmail(@RequestParam String email) {
+        return userService.getByEmail(email);
+    }
+
+    //for admin only
+    @GetMapping("/all")
```

Comment:
Avoid using words such as `all`, `update` etc in endpoint naming as this can be replaced with specific methods
So `GET` `/users` will signify that it's used to get all users 
whereas `POST` `/users` will be used to create user
`GET` `/users/{id}` will be used to get one user
`PUT` `/users/{id}` to update and so on...